### DISCUSSION TITLE
Tips for a better Git commit msg

### DISCUSSION THREAD

**Discussion by hodahakim** [NRTC/quick-start-guide#81] [Team Posts]
<div align="right">
سلام دوستان
‫برای نوشتن commit message یه همچین قواعدی وجود داره:‬
</div>
<div align="left">

https://chris.beams.io/posts/git-commit/
https://wiki.openstack.org/wiki/GitCommitMessages#Examples_of_good_practice_2

</div>
<div align="right">
‫این هم نسخه خلاصه تر:‬
</div>
<div align="left">

https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

</div>
<div align="right">
‫و جالبه که میشه پیام  commit های قبلی رو هم تغییر داد:‬
</div>
<div align="left">

https://help.github.com/en/articles/changing-a-commit-message
https://gist.github.com/nepsilon/156387acf9e1e72d48fa35c4fabef0b4

</div>

**Comment by firouzian** [comment]
‫البته تا وقتی پوش نشده می‌شه تغییر دارد.‬

**Comment by hodahakim** [comment]
<div align="right">

این طور نیست، اصلا برای همین جالب بود که بعد از پوش کردن می شد تغییرش داد!ا

</div>


<div align="left">

https://help.github.com/en/articles/changing-a-commit-message#amending-older-or-multiple-commit-messages
https://gist.github.com/nepsilon/156387acf9e1e72d48fa35c4fabef0b4#already-pushed--old-commit

</div>

**Comment by firouzian** [comment]
‫ممنون از شما. حق به جانب شماست. درواقع آپشن force امکان عوض کردن تاریچه را می‌دهد و برای همین می‌توان از آن استفاده کرد. امّا مشکلاتی هم دارد:‬

We strongly discourage force pushing, since this changes the history of your repository. If you force push, people who have already cloned your repository will have to manually fix their local history. For more information, see "Recovering from upstream rebase" in the Git manual.
